disnake/onboarding.py

```diff
@@ -185,12 +205,37 @@ def __repr__(self) -> str:
             f"description={self.description!r} emoji={self.emoji!r}>"
         )
 
-    @property
-    def roles(self) -> List[Role]:
-        """List[:class:`Role`]: A list of roles that will be added to the user when they select this option."""
-        return list(filter(None, map(self.guild.get_role, self.role_ids)))
-
-    @property
-    def channels(self) -> List[GuildChannel]:
-        """List[:class:`abc.GuildChannel`]: A list of channels that the user will see when they select this option."""
-        return list(filter(None, map(self.guild.get_channel, self.channel_ids)))
+    @classmethod
+    def _from_dict(cls, *, data: OnboardingPromptOptionPayload, guild: Guild) -> Self:
```

Comment:
This doesn't need a `Guild` object right now, passing the state should be enough

Reply:
👍59b49d69aea32314e84acf9b9584ab82b4d7f948